evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py

```diff
@@ -40,9 +40,15 @@ def __init__(self, boosting_type="gbdt", learning_rate=0.1, n_estimators=100, ma
                       "max_depth": max_depth,
                       "num_leaves": num_leaves,
                       "min_child_samples": min_child_samples,
-                      "n_jobs": n_jobs}
+                      "n_jobs": n_jobs,
+                      "bagging_freq": bagging_freq,
+                      "bagging_fraction": bagging_fraction}
```

Comment:
@bchen1116 could you please explain why adding these two parameters fixed the bug?

Reply:
As some background, LightGBM has 4 boosting types: "gbdt", "dart", "goss", "rf". `Bagging_freq` refers to the frequency of bagging, where it bags every `bagging_freq = k` iterations (0 means it doesn't bag). `bagging_fraction` refers to the amount of data randomly selected without resampling (1 means select all, 0 means none). This can help speed up the training process. 

The default `bagging_freq` that LightGBM sets is 0, which works with `gbdt`, `dart`, and `goss`. However, for `rf`, since its random forest, LightGBM requires that it uses bagging, which means `bagging_freq` must be 1 and `bagging_fraction` must be set to be below 1.0. By adding those two parameters and changing `bagging_freq` when the `boosting_type=rf`, we do a simple fix to avoid this bug.

Reply:
Thanks for the clear explanation! That makes sense.

Can we tweak the comment you left on line 48:
> if the boosting type is random forest, bagging is required by lightgbm, so we set bagging_freq to 1 in order to avoid errors